packages/gas-fee-controller/src/fetchBlockFeeHistory.ts

```diff
@@ -214,11 +216,13 @@ function buildExistingFeeHistoryBlock<Percentile extends number>({
 }): ExistingFeeHistoryBlock<Percentile> {
   const gasUsedRatio = gasUsedRatios[blockIndex];
   const priorityFeesForEachPercentile = priorityFeePercentileGroups[blockIndex];
-  const priorityFeesByPercentile = percentiles.reduce(
+  const priorityFeesByPercentile = percentiles.reduce<Record<Percentile, BN>>(
     (obj, percentile, percentileIndex) => {
       const priorityFee = priorityFeesForEachPercentile[percentileIndex];
       return { ...obj, [percentile]: fromHex(priorityFee) };
     },
+    // Without this, the return type of `reduce` is inferred improperly.
```

Comment:
Yes, okay. I knew there was a reason why I preferred typecasting the accumulator argument rather than specifying the type parameter (even if it's less safe). Do you remember what the inferred type was here?

Reply:
It's inferred as the generic `Percentile`. 🤔